HyprMXAdapter/src/main/java/com/chartboost/mediation/hyprmxadapter/HyprMXAdapter.kt

```diff
@@ -756,98 +747,43 @@ class HyprMXAdapter : PartnerAdapter {
-                        ad = placement,
-                        details = emptyMap(),
-                        request = request,
+        override fun onAdLoaded(isAdAvailable: Boolean) {
+            if (isAdAvailable) {
+                PartnerLogController.log(LOAD_SUCCEEDED)
+                resumeOnce(
+                    Result.success(
+                        PartnerAd(
+                            ad = placement,
+                            details = emptyMap(),
+                            request = request,
+                        ),
                     ),
-                ),
-            )
-        }
-
-        override fun onAdNotAvailable(placement: Placement?) {
-            PartnerLogController.log(LOAD_FAILED)
-            resumeOnce(
-                Result.failure(
-                    ChartboostMediationAdException(
-                        ChartboostMediationError.CM_LOAD_FAILURE_NO_FILL,
+                )
+            } else {
+                PartnerLogController.log(LOAD_FAILED)
+                resumeOnce(
+                    Result.failure(
+                        ChartboostMediationAdException(
+                            ChartboostMediationError.CM_LOAD_FAILURE_NO_FILL,
+                        ),
                     ),
-                ),
-            )
+                )
+            }
         }
     }
 
-    /**
-     * Callback class for rewarded ads.
-     *
-     * @param continuationRef A [WeakReference] to the [CancellableContinuation] to be resumed once the ad is shown.
-     * @param request A [PartnerAdLoadRequest] object containing the request.
-     * @param listener A [PartnerAdListener] to be notified of ad events.
-     */
-    private class RewardedAdListener(
-        private val continuationRef: WeakReference<CancellableContinuation<Result<PartnerAd>>>,
+    private class ShowAdListener(
         private val request: PartnerAdLoadRequest,
-        private val listener: PartnerAdListener,
-    ) : RewardedPlacementListener {
-
-        fun resumeOnce(result: Result<PartnerAd>) {
-            continuationRef.get()?.let {
-                if (it.isActive) {
-                    it.resume(result)
-                }
-            } ?: run {
-                PartnerLogController.log(LOAD_FAILED, "Unable to resume continuation. Continuation is null.")
-            }
-        }
-
-        override fun onAdStarted(placement: Placement?) {
+    ) : HyprMXShowListener {
+        override fun onAdStarted(placement: Placement) {
             PartnerLogController.log(SHOW_SUCCEEDED)
             onShowSuccess()
             onShowSuccess = {}
         }
 
-        override fun onAdClosed(
-            placement: Placement?,
-            finished: Boolean,
-        ) {
+        override fun onAdClosed(placement: Placement, finished: Boolean) {
             PartnerLogController.log(DID_DISMISS)
-            listener.onPartnerAdDismissed(
+            request.adInteractionListener.onDismissed(
```

Comment:
unfortunately, this won't build for 4.0. This was added to the partner load request for adaptive banners and is not available until 4.6 or so.
you're going to have to store the original partner ad listener somewhere and pass it to each load somehow. We typically have been storing them in maps keyed by request ID, but it's gotta live somewhere.

Reply:
Ok, makes sense.  And I found an example to replicate.